src/test/e2e/api/client/global.segment.e2e.test.ts

```diff
@@ -0,0 +1,139 @@
+        .then((res) => res.body.features);
+};
+
+const fetchClientResponse = (): Promise<ApiResponse> => {
+    return app.request
+        .get(FEATURES_CLIENT_BASE_PATH)
+        .expect(200)
+        .then((res) => res.body);
+};
+
+const createSegment = (postData: object): Promise<unknown> => {
+    const user = { email: 'test@example.com' } as User;
+    return app.services.segmentService.create(postData, user);
+};
+
+const createFeatureToggle = (
+    postData: object,
+    expectStatusCode = 201,
+): Promise<unknown> => {
+    return app.request
+        .post(FEATURES_ADMIN_BASE_PATH)
+        .send(postData)
+        .expect(expectStatusCode);
+};
+
+const addSegmentToStrategy = (
+    segmentId: number,
+    strategyId: string,
+): Promise<unknown> => {
+    return app.services.segmentService.addToStrategy(segmentId, strategyId);
+};
+
+const mockFeatureToggle = (): object => {
+    return {
+        name: randomId(),
+        strategies: [{ name: randomId(), constraints: [], parameters: {} }],
+    };
+};
+
+const mockConstraints = (): IConstraint[] => {
+    return Array.from({ length: 5 }).map(() => ({
+        values: ['x', 'y', 'z'],
+        operator: 'IN',
+        contextName: 'a',
+    }));
+};
+
+const createTestSegments = async (): Promise<void> => {
+    const constraints = mockConstraints();
+    await createSegment({ name: 'S1', constraints });
+    await createSegment({ name: 'S2', constraints });
+    await createSegment({ name: 'S3', constraints });
+    await createFeatureToggle(mockFeatureToggle());
+    await createFeatureToggle(mockFeatureToggle());
+    await createFeatureToggle(mockFeatureToggle());
+    const [feature1, feature2] = await fetchFeatures();
+    const [segment1, segment2] = await fetchSegments();
+
+    await addSegmentToStrategy(segment1.id, feature1.strategies[0].id);
+    await addSegmentToStrategy(segment2.id, feature1.strategies[0].id);
+    await addSegmentToStrategy(segment2.id, feature2.strategies[0].id);
+};
+
+beforeAll(async () => {
+    const experimentalConfig = {
+        segments: {
+            enableSegmentsAdminApi: true,
+            enableSegmentsClientApi: true,
+            inlineSegmentConstraints: false,
+        },
+    };
+
+    db = await dbInit('global_segments', getLogger, {
+        experimental: experimentalConfig,
+    });
+
+    app = await setupAppWithCustomConfig(db.stores, {
+        experimental: experimentalConfig,
+    });
+});
+
+afterAll(async () => {
+    await app.destroy();
+    await db.destroy();
+});
+
+afterEach(async () => {
+    await db.stores.segmentStore.deleteAll();
+    await db.stores.featureToggleStore.deleteAll();
+});
+
+test('should return segments in base of toggle response if inline is disabled', async () => {
+    await createTestSegments();
+
+    const clientFeatures = await fetchClientResponse();
+    expect(clientFeatures.segments.length).toBeDefined();
+});
+
+test('should only send segments that are in use', async () => {
```

Comment:
Should we add a test for the `feature.segments` field as well?

Reply:
Yeah, I kinda started to do that but given that nothing in this project really understands what the segment is really, it became a test to check that what was put in was what came out, struggled to imagine what that test case would actually catch. Sounds like you might have an idea on a valuable test here, would love to hear a suggestion!

Reply:
Good point! I was thinking it would be nice to test that the presence of segments won't mess up the join/reduce logic in `src/lib/db/feature-toggle-client-store.ts`. But I suppose other tests might cover that already. :thinking:

Reply:
Ah, that's a good point, so it's a bit hard to see from the PR but I've collapsed some code branches here, I cannibalised a lot of the logic for inline segments so it's pretty well tested by previous tests. What isn't tested is that segments resolve in the way they should, i.e. the correct segment ids are present on the toggle and that maps nicely back to a global segment, I've added a test for that now, which definitely needed to be there